src/nwm_routing/src/nwm_routing/__main__.py

```diff
@@ -486,17 +494,30 @@ def main_v02(argv):
     data_assimilation_csv = data_assimilation_parameters.get(
         "data_assimilation_csv", None
     )
-    data_assimilation_filter = data_assimilation_parameters.get(
-        "data_assimilation_filter", None
+    data_assimilation_folder = data_assimilation_parameters.get(
+        "data_assimilation_folder", None
     )
-    if data_assimilation_csv or data_assimilation_filter:
+    data_assimilation_filter = data_assimilation_parameters.get("data_assimilation_filter",None)
+    last_obs_file = data_assimilation_parameters.get("wrf_hydro_last_obs_file", None)
+
+    last_obs_df = pd.DataFrame()
+
+    if last_obs_file:
+        usgs_df, last_obs_df, last_obs_date = nnu.build_data_assimilation(
+            data_assimilation_parameters
+        )
+        last_obs_start = nhd_io.get_last_obs_location(qlats, last_obs_date)
+
+    else:
```

Comment:
Are there corner cases for this logic? (i.e., `data_assimilation_filter == true` and `last_obs_file == false` with some condition that requires a 'last_obs` to jumpstart the data assimilation....)